.github/workflows/v2-tests-e2e.yml

```diff
@@ -59,22 +61,33 @@ jobs:
           install-command: yarn
       - name: Setup test-website project against master release
         run: |
-          KEEP_CONTAINER=true yarn test:build:v2
-          rm -rf node_modules
-      - name: Setup test-website project for Yarn v2
+          KEEP_CONTAINER=true yarn test:build:v2 -s
+      - name: Setup test-website project for Yarn berry with nodeLinker = ${{ matrix.nodeLinker }}
         run: |
-          cd test-website
+          mv test-website ../test-website
+          cd ../test-website
+
           yarn set version berry
+          yarn --version
+          yarn set version canary
+          yarn --version
+
+          yarn config set nodeLinker ${{ matrix.nodeLinker }}
           yarn config set pnpMode loose
```

Comment:
```suggestion
```
There is an undeclared dependency error that this is allowing to pass

Reply:
Yes I've seen those.
However some are:
- mdx loader (my fault, will fix)
- `debug` using `supports-color` (not my fault, how to fix?)
- code copied from CRA using Node.js `url`. We'll likely delete this code and move back to using CRA code (which also doesn't declare `url` as dependency btw). Also I'm not totally sure how to fix this. Should I just add npm package `url` as dependency? How can I know the good version to use considering it's provided by Node.js runtime?

Reply:
I'd like to fix those but likely in another PR. 
The goal here is just to make our existing CI pass again (which was already using loose mode)

Reply:
> debug using supports-color (not my fault, how to fix?)

Nothing to fix, it's an optional peer dependency that isn't provided by the parent so it logs an error but it's fine to ignore and wont be visible in strict mode

> code copied from CRA using Node.js url. We'll likely delete this code and move back to using CRA code (which also doesn't declare url as dependency btw). Also I'm not totally sure how to fix this. Should I just add npm package url as dependency? How can I know the good version to use considering it's provided by Node.js runtime?

It's because Webpack 5 doesn't polyfill Node libraries by default, you need to do it yourself

Reply:
> Nothing to fix, it's an optional peer dependency that isn't provided by the parent so it logs an error but it's fine to ignore and wont be visible in strict mode

Thanks

> It's because Webpack 5 doesn't polyfill Node libraries by default, you need to do it yourself

I've tried adding `alias: {url: 'url'}` in Webpack config but the error was still there. It's not clear to me what I should do exactly and the Medium link you gave in some yarn issue didn't make it very clear either

Reply:
You're running into https://github.com/webpack/changelog-v5/tree/b9a346f9090bf2e9f8bc96864a88ed07ad7bc1a4#automatic-nodejs-polyfills-removed so aliasing `url` to `url` makes no difference.

Next.js deals with this by aliasing `url` to `native-url` https://github.com/vercel/next.js/blob/681d298bdf7048bdbd2790a551d5d231906ede6f/packages/next/build/webpack-config.ts#L125

Reply:
I see thanks, will try that. 
For the `mdx-loader` issue, unfortunately I can't just add it because it would create a cycle in Lerna build:

> lerna WARN ECYCLE @docusaurus/mdx-loader -> @docusaurus/core -> @docusaurus/mdx-loader

I see how I could re-org the code to avoid this cycle but was wondering if there's a way to just tell Yarn that the dependency is here without having to declare it in any deps? `peerDependency` works but is annoying because all site's parent lockfile would now have to declare `@docusaurus/mdx-loader`, while we know for sure it is here.